src/test/java/com.gigssandbox/UserServiceTest.java

```diff
@@ -39,45 +39,45 @@ public void shoudAddUserToUsersCollectionWhenUserTriesToRegister() {
     @Test
     public void shouldThrowAlreadyRegisteredExceptionWhenUserCredentialsAreAlreadyPresentInUsersMap() {
         String username = "skip";
-        char[] password = "the foreplay".toCharArray();
-        Whitebox.setInternalState(userService, "users", Collections.singletonMap(username, User.builder().username(username).passwordHash(Arrays.hashCode(password)).build()));
+        String password = "the foreplay";
+        Whitebox.setInternalState(userService, "users", Collections.singletonMap(username, User.builder().username(username).passwordHash(password.hashCode()).build()));
 
         assertThrows(AlreadyRegisteredException.class, () -> userService.registerUser(username, password));
     }
 
     @Test
     public void shouldThrowUserNotRegisteredExceptionWhenUserCredentialsAreNotPresentInUsersMap() {
         String username = "ben_bruice";
-        char[] password = "everyday00playing00zeros".toCharArray();
+        String password = "everyday00playing00zeros";
 
         assertThrows(NotRegisteredException.class, () -> userService.logUserIn(username, password));
     }
 
     @Test
     public void shouldThrowIncorrectPasswordExceptionWhenUserPassedIncorrectPassword() {
```

Comment:
This one is better, but lets shouldThrow without exception specific name. IncorrectPasswordException.class name can be refactored later and you will have to refactor test name too. Lets avoid it.

```
@Test(expected = IncorrectPasswordException.class)
shouldThrowWhenIncorrectUserPasswordIsPassed {
...
```